crates/polars-stream/src/async_primitives/distributor_channel.rs

```diff
@@ -198,7 +199,7 @@ impl<T: Send> Sender<T> {
 }
 
 impl<T: Send> Receiver<T> {
-    pub async fn recv(&mut self) -> Result<T, ()> {
+    pub async fn recv(&self) -> Result<T, ()> {
```

Comment:
I made this non-`mut` as I needed to re-use the receivers across execution phases in `ParquetSourceNode::spawn`, and this was the easiest way, but I'm not sure if it's a good thing to do.

Reply:
Please undo this change, this makes it unsound. The receiver must be single-threaded.

Reply:
You can just wrap the receivers in a `Vec<Mutex<Receiver>>` if necessary, and lock the mutex for the relevant receiver once in each task before the main loop starts.

Reply:
I see - I have reverted it and left a comment